src/Microsoft.AspNet.OData/EnableQueryAttribute.cs

```diff
@@ -71,6 +73,34 @@ public override void OnActionExecuted(HttpActionExecutedContext actionExecutedCo
                 throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveActionContext);
             }
 
+            bool hasExpandQuery = false;
+            string queryString = request.RequestUri.Query;
+            var absoluteUri = request.RequestUri.AbsoluteUri;
+            bool hasEmptyQueryString = string.IsNullOrEmpty(queryString);
+
+            ReadOnlySpan<char> s = absoluteUri.AsSpan();
+            string baseUriWithPath = hasEmptyQueryString ? absoluteUri : s.Slice(0, absoluteUri.IndexOf('?', 0)).ToString();
+
+            if (!hasEmptyQueryString)
+            {
+                var queries = queryString.Split('&');
+
+                hasExpandQuery = queries.Any(x => x.StartsWith("?$expand", StringComparison.OrdinalIgnoreCase) || x.StartsWith("$expand", StringComparison.OrdinalIgnoreCase) || x.StartsWith("expand", StringComparison.OrdinalIgnoreCase));
+            }
+
+            // Create a $expand query string if 1) It's a POST request 2) if there is no expand query string.
+            if (String.Equals(actionExecutedContext.Request.Method, HttpMethod.Post) && !hasExpandQuery)
+            {
+                string expand = GenerateExpandQueryFromPayload(actionExecutedContext.ActionContext);
+
+                if (!string.IsNullOrEmpty(expand))
+                {
+                    // If query string was empty, we add a new query string e.g ?$expand=Order. If not, we prepend a $expand query
+                    expand = hasEmptyQueryString ? "?" + expand : "?" + expand + "&" + queryString.TrimStart('?');
```

Comment:
Is there any specific reason that makes it necessary for `$expand` query option to be at the start of the query string?

Reply:
No specific reason. It's just easier to append it at the front